gcloud/bigquery/table.py

```diff
@@ -693,6 +699,225 @@ def insert_data(self,
 
         return errors
 
+    @staticmethod
+    def _configure_job_metadata(metadata,  # pylint: disable=R0913
+                                allow_jagged_rows,
+                                allow_quoted_newlines,
+                                create_disposition,
+                                encoding,
+                                field_delimiter,
+                                ignore_unknown_values,
+                                max_bad_records,
+                                quote_character,
+                                skip_leading_rows,
+                                write_disposition):
+        """Helper for :meth:`upload_from_file`."""
+        load_config = metadata['configuration']['load']
+
+        if allow_jagged_rows is not None:
+            load_config['allowJaggedRows'] = allow_jagged_rows
+
+        if allow_quoted_newlines is not None:
+            load_config['allowQuotedNewlines'] = allow_quoted_newlines
+
+        if create_disposition is not None:
+            load_config['createDisposition'] = create_disposition
+
+        if encoding is not None:
+            load_config['encoding'] = encoding
+
+        if field_delimiter is not None:
+            load_config['fieldDelimiter'] = field_delimiter
+
+        if ignore_unknown_values is not None:
+            load_config['ignoreUnknownValues'] = ignore_unknown_values
+
+        if max_bad_records is not None:
+            load_config['maxBadRecords'] = max_bad_records
+
+        if quote_character is not None:
+            load_config['quote'] = quote_character
+
+        if skip_leading_rows is not None:
+            load_config['skipLeadingRows'] = skip_leading_rows
+
```

Comment:
With all these `not None` checks ISTM we'd be better off using `kwargs` and just using the docstring to indicate which ones we want. This would also eliminate the need for increasing the number of arguments.